xformers/components/attention/csrc/cuda/mem_eff_attention/kernel_backward.h

```diff
@@ -1348,12 +1348,14 @@ struct AttentionBackwardKernel {
     bool rowPred = (query_start + laneRow) < p.num_queries;
     bool pred = rowPred;
 
-    const __restrict__ AccessType* grad_output_ptr =
-        reinterpret_cast<const __restrict__ AccessType*>(
+    // on windows, previous syntax __restrict__ AccessType*
+    // resulted in error: "restrict" is not allowed 
+    const AccessType* __restrict__ grad_output_ptr =
+        reinterpret_cast<const AccessType* __restrict__>(
             p.grad_output_ptr + (query_start + laneRow) * p.head_dim_value +
             laneFirstCol);
-    const __restrict__ AccessType* output_ptr =
-        reinterpret_cast<const __restrict__ AccessType*>(
+    const AccessType* __restrict__ output_ptr =
+        reinterpret_cast<const AccessType* __restrict__>(
```

Comment:
@danthe3rd looks like we might have switched the order of the `restrict` keywords here. Do you think the `__restrict__` keyword was being picked by the compiler before in here?

Reply:
that's a good point. I'll run some benchmarks after that